src/systems.rs

```diff
@@ -148,3 +148,44 @@ impl<'a> System<'a> for AttackSystem {
         }
     }
 }
+
+pub struct SpawnerSystem;
+
+impl<'a> System<'a> for SpawnerSystem {
+    type SystemData = (
+        Read<'a, DeltaTime>,
+        Entities<'a>,
+        Read<'a, LazyUpdate>,
+        ReadStorage<'a, Transform>,
+        WriteStorage<'a, Spawner>,
+    );
+
+    fn run(&mut self, data: Self::SystemData) {
+        let (dt, entities, lazy, transforms, mut spawners) = data;
+
+        for (transform, spawner, spawner_ent) in (&transforms, &mut spawners, &entities).join() {
+            if spawner.cooldown > 0.0 {
+                spawner.cooldown -= dt.0;
+            } else {
+                spawner.cooldown = spawner.seconds_to_spawn;
+                // Spawn entity
+                let new_ent = entities.create();
+                lazy.insert(new_ent, *transform);
+                lazy.insert(new_ent, Velocity::new(60.0, 60.0));
+                lazy.insert(new_ent, Drawable::Enemy);
+                lazy.insert(new_ent, Collider::new(40.0, 40.0));
+                lazy.insert(new_ent, Health {current_hp: 5});
+                lazy.insert(new_ent, spawner.spawn_faction);
+                //TODO - why does this give an issue?
+                //lazy.insert(new_ent, spawner.spawn_drawable);
```

Comment:
Woops, just noticed this. Looks like it's complaining due to borrow stuff! You can fix it by doing `spawner.spawn_drawable.clone()`. Or we could derive `Copy` on `Drawable`. Either one sounds good to me.

Reply:
Sweet! I'll derive Copy and then commit/merge :)